File: Data Structures & Algorithms/daily-temperatures/submission-1.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<int[]> st = new Stack<>(); //(temp, index)
        int n = temperatures.length;
        int[] ans = new int[n];
        for(int i = n-1;i>=0;i--){
            int count = 0;
            while(!st.isEmpty() && st.peek()[0] <= temperatures[i]){
                st.pop();
            }
            
            ans[i] = !st.isEmpty() ? st.peek()[1]-i : 0;
            st.push(new int[]{temperatures[i],i});
        }
        return ans;
    }
}

File: Data Structures & Algorithms/climbing-stairs/submission-1.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 0 or n == 1:
            return n
        # ans = [0]*(n+1)
        # print(ans)
        first = 1
        second = 1
        for i in range(2,n+1):
            temp = first + second
            first = second
            second = temp
        # print(ans)
        return second
        